### DISCUSSION TITLE
Add L/s (litres per second) as a valid unit of measurement to volume_flow_rate device class

### DISCUSSION THREAD

**Discussion by bradkeifer** [home-assistant/architecture#1044] [Entity Models] [answered]
The volume_flow_rate device class currently does not support a litres persecond unit of measurement.

I have a requirement for a sensor and also dashboard display of volume flow rates in litres per second (L/s). I also note that the SI unit of measurement for volume flow rate is cubic metres per second and this is also not supported. I make use of the modbus integration to obtain my sensor data, and I was alerted to the lack of support for the L/s unit of measurement when reviewing the home-assistant.log file at startup, where the following warning message was given:
```
2024-02-21 10:32:19.879 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.d1_flow (<class 'homeassistant.components.modbus.sensor.ModbusRegisterSensor'>) is using native unit of measurement 'l/s' which is not a valid unit for the device class ('volume_flow_rate') it is using; expected one of ['m³/h', 'gal/min', 'L/min', 'ft³/min']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+modbus%22
```
The configuration I have for this sensor is:
```
- name: D1 Flow
        unit_of_measurement: "l/s"
        device_class: volume_flow_rate
        address: 29689
        scale: 1
        precision: 0
        data_type: uint32
```

I would like to see litres per second (L/s) and possibly cubic metres per second (m³/s) added to the set of units_of_measurement for the volume_flow_rate device class please.

**Comment by MartinHjelmare** [comment]
Sounds ok to add support for litre per second.

Cubic meter per second doesn't sound like it would be used in a home.

**Reply by elupus** [reply]
L/h would be good as well. Its the native unit of knx.

**Reply by bradkeifer** [reply]
> Cubic meter per second doesn't sound like it would be used in a home.

I'm not sure what the architectural guidelines are here, but cubic metres per second is the SI unit of measurement for volume flow rate. I agree that cubic metres per second would be unusual in a home context.

**Reply by bradkeifer** [reply]
@MartinHjelmare - what is the best way to move forward with this? I am happy to have a go at making the changes and submitting them for review and inclusion in a future release, but I am not sure of the correct process to follow.

**Reply by MartinHjelmare** [reply]
We'll only add units that has a common application in Home Assistant. We don't add units just because they are part of the SI system.

**Reply by patrickvorgers** [reply]
For residential usage L/h is much more common than m3/h. In my case I have a heatpump which shows the flowrate in L/h.
Has there been an architectural decision whether L/h and/or L/s will be supported?

**Reply by mib1185** [reply]
the `pegel_online` integration would benefit from `m³/s` - see https://github.com/home-assistant/core/pull/143563#discussion_r2058198870

**Reply by mib1185** [reply]
@MartinHjelmare would you agree with adding `l/s` , `l/h` and `m³/s` ?

**Reply by MartinHjelmare** [reply, marked as answer]
Yes, all three are ok add. We're now accepting new units where there's a simple conversion for existing device classes without architecture discussion.

**Reply by mib1185** [reply]
https://github.com/home-assistant/core/pull/143625
